Samples/AppLifecycle/Restart/cpp-console-unpackaged/cpp-console-unpackaged.cpp

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include <iostream>
+#include <string>
+
+#include <wil/result.h>
+#include <wil/cppwinrt.h>
+
+#include <winrt/Windows.ApplicationModel.Background.h>
+#include <winrt/Windows.ApplicationModel.Activation.h>
+#include <winrt/Windows.Foundation.h>
+#include <winrt/Microsoft.Windows.AppLifecycle.h>
+#include <winrt/Windows.Foundation.Collections.h>
+#include <winrt/Windows.Storage.h>
+
+
+using namespace winrt::Microsoft::Windows::AppLifecycle;
+using namespace winrt::Windows::ApplicationModel::Activation;
+using namespace winrt::Windows::ApplicationModel::Core;
+using namespace winrt::Windows::Foundation;
+
+void printArguments()
+{
+    AppActivationArguments args = AppInstance::GetCurrent().GetActivatedEventArgs();
+    switch (args.Kind())
+    {
+        case ExtendedActivationKind::Launch:
+        {
+            ILaunchActivatedEventArgs launchArgs = args.Data().as<ILaunchActivatedEventArgs>();
+            if (launchArgs)
+            {
+                winrt::hstring argString = launchArgs.Arguments();
+                std::cout << "Command line launch arguments:\n" << winrt::to_string(argString) << std::endl;
+            }
+        }
+    }
+}
+
+void restartApp()
+{
+    std::string restartArgs;
+
+    std::cout << "Enter restart arguments: ";
+    std::cin.ignore();
+    getline(std::cin, restartArgs);
+
+    AppRestartFailureReason restartError{ AppInstance::Restart(winrt::to_hstring(restartArgs)) };
+
+    switch (restartError)
+    {
+        case AppRestartFailureReason::RestartPending:
+            std::cout << "Another restart is currently pending.";
+            break;
+        case AppRestartFailureReason::InvalidUser:
+            std::cout << "Current user is not signed in or not a valid user.";
```

Comment:
This seems like a curious case - when would this happen? ...and is there any other reason you'd get InvalidUser?

Reply:
Because we leverage requestrestartasync, which _does_ return the invalid user failure reason, we have to catch this case as well.